### DISCUSSION TITLE
I'm getting eslint errors when using auto-import inside composables

### DISCUSSION THREAD

**Discussion by stefanobartoletti** [nuxt/nuxt#22087] [Questions]
I'm getting eslint errors when I use autoimporting of Vue core functions and of other composables inside a custom composable.

In example, if I create this composable 

```js
export const useMyComposable = () => {
  const num = ref(0)

  onMounted(() => {
    console.log('mounted. ref is ' + num.value)
  })

  onUnmounted(() => {
    console.log('unmounted. ref is ' + num.value)
  })

  return {
    num,
  }
}

```

eslint is throwing me these errors

```
  2:15  error  'ref' is not defined          no-undef
  4:3   error  'onMounted' is not defined    no-undef
  8:3   error  'onUnmounted' is not defined  no-undef
```

while, in reality, the code runs fine and as expected, using this composable in one of my components is not giving any error and produces the expected output.

These eslint errors appear only when trying to use auto-imports (both of core Vue functions and of other composables) inside composables, while auto-imports inside components are working as expected and not reported as errors.

Can someone help me understand this?

**Comment by JCtapuk** [comment]
> I'm getting eslint errors when I use autoimporting of Vue core functions and of other composables inside a custom composable.
> 
> In example, if I create this composable
> 
> ```js
> export const useMyComposable = () => {
>   const num = ref(0)
> 
>   onMounted(() => {
>     console.log('mounted. ref is ' + num.value)
>   })
> 
>   onUnmounted(() => {
>     console.log('unmounted. ref is ' + num.value)
>   })
> 
>   return {
>     num,
>   }
> }
> ```
> 
> eslint is throwing me these errors
> 
> ```
>   2:15  error  'ref' is not defined          no-undef
>   4:3   error  'onMounted' is not defined    no-undef
>   8:3   error  'onUnmounted' is not defined  no-undef
> ```
> 
> while, in reality, the code runs fine and as expected, using this composable in one of my components is not giving any error and produces the expected output.
> 
> These eslint errors appear only when trying to use auto-imports (both of core Vue functions and of other composables) inside composables, while auto-imports inside components are working as expected and not reported as errors.
> 
> Can someone help me understand this?

`import { ref, useMounted, useUnmounted } from '#imports'`

**Comment by stefanobartoletti** [comment]
Thanks for the reply, but it does not answer to what I was asking.
I don't need to edit my source code to make it work, it is already fully functional, I only have a problem with eslint that marks these auto-imports as errors

**Reply by wesley3295** [reply]
Did you ever find a solution?
